### PR TITLE
chore: update docs for shared subscribe

### PR DESCRIPTION
**Pull request by hjianbo**

### CODE REVIEW

File: en_US/configuration/configuration.md [hunk 1]
```diff
@@ -4009,17 +4009,34 @@ Set the type of session cluster lock. The session cluster lock is used to preven
...
+
+<br />
+
+### broker.sample_group.shared_subscription_strategy
+
+| Type | Optional Value                                                   | Default  |
+| ---- | ---------------------------------------------------------------- | -------- |
+| enum | `hash_clientid`, `hash_topic`, `local`, `random`, `round_robin`, `sticky`,  | - |
+
+Overrides the dispatch strategy for a shared subscription group named `sample_group`. If not configured, `broker.shared_subscription_strategy` will be used.
```
Comment (zmstone) [inline en_US/configuration/configuration.md]:
```suggestion
Overrides the dispatch strategy for a shared subscription group named `sample_group`. If not configured, `broker.shared_subscription_strategy` is used,
```

File: en_US/configuration/configuration.md [hunk 2]
```diff
@@ -4009,17 +4009,34 @@ Set the type of session cluster lock. The session cluster lock is used to preven
...
+
+Where `sample_group` can be configured to any group name.
```
Comment (zmstone) [inline en_US/configuration/configuration.md:4035]:
```suggestion
Where `sample_group` can be configured to any group name.
```

File: en_US/configuration/configuration.md [hunk 3]
```diff
@@ -4009,17 +4009,34 @@ Set the type of session cluster lock. The session cluster lock is used to preven
...
-| enum | `random`, `round_robin`, `sticky`, `hash_clientid`, `hash_topic` | `random` |
+| enum | `hash_clientid`, `hash_topic`, `local`, `random`, `round_robin`, `sticky`,  | - |
 
 #### Description
 
 Set a dispatch strategy for shared subscribers. Options are:
 
+- **hash_clientid**: Map (hash) publisher ClientID to subscriber.
+- **hash_topic**: Map (hash) source MQTT topic to subscriber.
+- **local**: Selects random subscriber connected to the node which received the message. If no such subscribers present, selects a random cluster-wise
```
Comment (zmstone) [inline en_US/configuration/configuration.md]:
```suggestion
- **local**: Select a random subscriber in the group locally in the node which is processing the shared-publish request. If there is no such local subscriber, it fallbacks to `random` strategy 
```

File: en_US/configuration/configuration.md (hunk 2 above)
Comment (hjianbo) [reply en_US/configuration/configuration.md:4035]: Looks the same? Did you miss something?

File: en_US/configuration/configuration.md [hunk 4]
```diff
@@ -4009,17 +4009,34 @@ Set the type of session cluster lock. The session cluster lock is used to preven
...
 #### Description
 
 Set a dispatch strategy for shared subscribers. Options are:
 
+- **hash_clientid**: Map (hash) publisher ClientID to subscriber.
+- **hash_topic**: Map (hash) source MQTT topic to subscriber.
+- **local**: Select a random subscriber in the group locally in the node which is processing the shared-publish request. If there is no such local subscriber, it fallbacks to `random` strategy 
 - **random**: Choose randomly among all subscribers.
 - **round_robin**: Enumerate subscribers in (unsorted) order.
 - **sticky**: First dispatch is random, then stick to it for all subsequent messages.
```
Comment (lafirest) [inline en_US/configuration/configuration.md]: maybe `The first dispatch of each session` is better?
Or will this https://github.com/emqx/emqx/issues/9062 be a bug?
The customers think the `sticky` is the same for the same `clientid` in different sessions

Comment (zmstone) [reply en_US/configuration/configuration.md]: > maybe `The first dispatch of each session` is better?
> Or will this [emqx/emqx#9062](https://github.com/emqx/emqx/issues/9062) be a bug?
> The customers think the `sticky` is the same for the same `clientid` in different sessions

Maybe add: “The stickiness lifecycle is bound to the publisher connection lifecycle, that is: if the publisher had to reconnect, stickiness will have to be reinitialized”

Comment (hjianbo) [reply en_US/configuration/configuration.md]: I corrected that desc. Plz check it🤔

File: en_US/advanced/shared-subscriptions.md [hunk 5]
```diff
@@ -67,18 +67,20 @@ broker.shared_subscription_strategy = random
 # Per-group balancing strategy
 broker.$group_name.shared_subscription_strategy = local
 
-# Applicable to QoS1 QoS2 messages, when enabled, message will be distributed to another group when one group is offline
+# Works for QoS1 QoS2 messages
+# When enabled, after a shared subscriber is disconnected (but the session is still stored in the server)
```
Comment (lafirest) [inline en_US/advanced/shared-subscriptions.md]: nit: If enabled, when a shared ...

**Review by lafirest — DISMISSED** [review]

**Review by terry-xiaoyu — APPROVED** [review]

**Review by zmstone — APPROVED** [review]

**Pull request merged**